org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalReplacementProvider.java

```diff
@@ -657,6 +672,7 @@ private void appendGuessingCompletion(StringBuilder buffer, CompletionProposal p
 				JavaLanguageServerPlugin.logException(e.getMessage(), e);
 			}
 		}
+		final var placeholderOffset = placeholderOffset(buffer);
```

Comment:
Is there ever a case where there's already a placeholder inserted into the buffer prior to this call ?

Reply:
When having a chain completion proposal with already placeholders in cases of a array chain node receiver which has gone through `ChainCompletionProposalComputer.appendArrayDimensions` we might have such a use case that needs to adjust the placeholder indexes right ?

Reply:
Yes! Sorry, you're right. We add a placeholder for the array dimension and if we add additional placeholders later, it would be good to track how many we have already added. It's probably a rarer case but good to fix.